File: Data Structures & Algorithms/count-connected-components/submission-0.java
class Solution {

    public int find(int x , int[]parent){
        if (parent[x] == x){
            return x;
        }
        else{
            return find(parent[x],parent);
        }
    }

    public void union(int a,int b,int[] parent){
        int parentA = find(a,parent);
        int parentB = find(b,parent);

        if(parentA!=parentB){
            parent[parentA] = parentB;
        }
    }

    public int countComponents(int n, int[][] edges) {
        int[] parent = new int[n];

        for(int i=0;i<n;i++){
            parent[i] = i;
        }

        for(int[]edge :edges){
            union(edge[0],edge[1],parent);
        }
        int count = 0;

        for(int i = 0;i<n;i++){
            if (parent[i]==i){
                count ++;
            }
        }

        return count;
    }


}

File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        map={}
        for num in nums:
            if(num in map):
                map[num]+=1
            else:
                map[num]=1
        
        for count in map.values():
            if count > 1:
                return True
        
        return False



            


            